oprf/oprf_test.go

```diff
@@ -0,0 +1,230 @@
+	}
+
+	if !(len(h) > 0) {
+		t.Fatal("invalid finalizing of client: no final byte array.")
+	}
+}
+
+func blindTest(c *group.Ciphersuite, in []byte) (*Token, BlindToken) {
+	bytes, _ := hex.DecodeString("e6db3004c35ec2cf97c4d462e4690e9859741c186b8e1138b977d547ad166951")
+	s := group.NewScalar(c)
+	s.Set(bytes)
+
+	p, _ := c.HashToGroup(in)
+	t := p.ScalarMult(s)
+	bToken := t.Serialize()
+
+	token := &Token{in, s}
+	return token, bToken
+}
+
+func TestClientBlindVector(t *testing.T) {
+	srv, err := NewServer(OPRFP256)
+	if err != nil {
+		t.Fatal("invalid setup of server: " + err.Error())
+	}
+	if srv == nil {
+		t.Fatal("invalid setup of server: no server.")
+	}
+
+	client, err := NewClient(OPRFP256)
+	if err != nil {
+		t.Fatal("invalid setup of client: " + err.Error())
+	}
+
+	_, bToken := blindTest(client.suite, []byte{00})
+
+	// From the test vectors
+	testBToken, _ := hex.DecodeString("02fb7eadba79acefca3e5401e291f2face38f4f3c159e8d636b29f650d96dfc3f1")
+	if !bytes.Equal(testBToken[:], bToken[:]) {
+		t.Errorf("blind elements are not equal: vectorToken: %x blindToken: %x", testBToken[:], bToken[:])
+	}
+}
+
+func TestServerEvaluationVector(t *testing.T) {
+	srv, err := NewServer(OPRFP256)
+	if err != nil {
+		t.Fatal("invalid setup of server: " + err.Error())
+	}
+	if srv == nil {
+		t.Fatal("invalid setup of server: no server.")
+	}
+	privKey, _ := hex.DecodeString("7331fb3bfbc4a786af3a35e33b2d75db3929b4c033998526dc66d60f6531a255")
+	srv.Keys.PrivK.Set(privKey)
+
+	client, err := NewClient(OPRFP256)
+	if err != nil {
+		t.Fatal("invalid setup of client: " + err.Error())
+	}
+
+	_, bToken := blindTest(client.suite, []byte{00})
+
+	eval, _ := srv.Evaluate(bToken)
+	if eval == nil {
+		t.Fatal("invalid evaluation of server: no evaluation.")
+	}
+
+	// From the test vectors
+	testEval, _ := hex.DecodeString("02449231545a1770f3e3995e7a0f0a29a51995bf068c833dd1269295641e289cda")
+
+	if !bytes.Equal(testEval[:], eval.element[:]) {
+		t.Errorf("eval elements are not equal: vectorEval: %x eval: %x", testEval[:], eval.element[:])
+	}
+}
+
+func TestClientUnblindVector(t *testing.T) {
+	srv, err := NewServer(OPRFP256)
+	if err != nil {
+		t.Fatal("invalid setup of server: " + err.Error())
+	}
+	if srv == nil {
+		t.Fatal("invalid setup of server: no server.")
+	}
+	privKey, _ := hex.DecodeString("7331fb3bfbc4a786af3a35e33b2d75db3929b4c033998526dc66d60f6531a255")
+	srv.Keys.PrivK.Set(privKey)
+
+	client, err := NewClient(OPRFP256)
+	if err != nil {
+		t.Fatal("invalid setup of client: " + err.Error())
+	}
+
+	token, bToken := blindTest(client.suite, []byte{00})
+
+	eval, _ := srv.Evaluate(bToken)
+	if eval == nil {
+		t.Fatal("invalid evaluation of server: no evaluation.")
+	}
+
+	info := []byte("test information")
+	iToken, h, _ := client.Finalize(token, eval, info)
+
+	// From the test vectors
+	testIToken, _ := hex.DecodeString("0277f2ded1d0cbbc9a71504f94dd0aa997709f7adb3368631392313164273eb340")
+
+	if !bytes.Equal(testIToken[:], iToken[:]) {
```

Comment:
not a must, but happy if the circl/internal/testing package is used for error reporting.

Reply:
Using it now ;)